python/kfserving/kfserving/storage.py

```diff
@@ -206,6 +211,18 @@ def _download_local(uri, out_dir=None):
             os.symlink(src, dest_path)
         return out_dir
 
+    @staticmethod
+    def _download_from_uri(uri, out_dir=None):
+        url = urlparse(uri)
+        fname = os.path.basename(url.path)
```

Comment:
should we validate whether this is `''` and fail loudly? E.g. if the url doesn't contain a valid file name?

Reply:
sgtm, I guess something like `http://foo.bar/test/` will also fail right?

Reply:
yes, it should, I'll add that along with an accompanying unit test.

Reply:
addressed in https://github.com/kubeflow/kfserving/pull/979/commits/00de564cb6a646146896946261a41d9699325a5a